content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -659,23 +664,32 @@ public void processUnsubmitted(Calendar cal) {
 				errors++;
 				continue;
 			}
-			// Get filename of submission
-			String fileName = resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME);
-			if (StringUtils.isEmpty(fileName)) {
-				// Set default file name:
-				fileName = "submission_" + item.getUserId() + "_" + item.getSiteId();
-				log.info("Using Default Filename " + fileName);
-			}
 			// EXTERNAL ID DOES NOT EXIST, CREATE SUBMISSION AND UPLOAD CONTENTS TO TCA
 			// (STAGE 1)
 			if (StringUtils.isEmpty(item.getExternalId())) {
+				// Get filename of submission
+				String fileName = resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME);
+				// If file name is missing extension add to fulfill TCA header requirement
+				if (FilenameUtils.getExtension(fileName).isEmpty()) {
+					// Add .html for in line submissions
+					if (INLINE_SUBMISSION.equals(fileName)) {
```

Comment:
This isn't a good check and makes assumptions on the title and the language. There is an "inline submission" flag property associated with the file itself:

ResourcePropertiesEdit inlineProps = contentHostingService.newResourceProperties();
        inlineProps.addProperty(ResourceProperties.PROP_DISPLAY_NAME, rb.getString("submission.inline"));
        inlineProps.addProperty(ResourceProperties.PROP_DESCRIPTION, resourceId);
        inlineProps.addProperty(AssignmentConstants.PROP_INLINE_SUBMISSION, "true");